Documentation/ExtensionArchitecture/FileStructure/Resources/Private/Index.rst

```diff
@@ -3,27 +3,149 @@
    Extension development; Resources/Private
    Folder; Resources/Private
    Resources; Private
-.. _extension-Resources-Private:
+.. _extension-resources-private:
 
-================================
-:file:`Private`
-================================
+=========
+`Private`
+=========
 
-.. index:: Path; EXT:{extkey}/Resources/Private/Layouts
+This folder contains resources of the that are needed when rendering a page
+but are not needed supplied directly to the browser. This includes:
 
-Resources/Private/Layouts
-  Main layouts for (Fluid) views.
+*   Fluid templates
+*   :ref:`Language files <extension-Resources-Private-Language>`
+*   Files for the compilation of assets like SCSS or TypeScript
 
-.. index:: Path; EXT:{extkey}/Resources/Private/Partials
+..  contents:: Table of contents
 
-Resources/Private/Partials
-  Partial templates for repetitive use.
+.. _extension-resources-private-fluid:
 
-.. index:: Path; EXT:{extkey}/Resources/Private/Templates
+Fluid templates in the folder Resources/Private
+===============================================
 
-Resources/Private/Templates
-  One template per action, stored in a folder named after each Controller.
+Fluid templates are commonly stored in a folder called
+:path:`Resources/Private/Templates`. The concrete location of templates is
+configurable via:
 
+*   TypoScript for `PAGEVIEW <https://docs.typo3.org/permalink/t3tsref:cobj-pageview>`_:
+    :ref:`paths.[priority] <t3tsref:confval-pageview-paths>`
+*   TypoScript for `FLUIDTEMPLATE <https://docs.typo3.org/permalink/t3tsref:cobj-template>`_:
+    :ref:`templateRootPaths <t3tsref:confval-fluidtemplate-templaterootpaths>` etc.
+*   Mails via global configuration:
+    :ref:`$GLOBALS['TYPO3_CONF_VARS']['MAIL']['templateRootPaths'] <t3coreapi:confval-globals-typo3-conf-vars-mail-templaterootpaths>`
+*   Settings of diverse system or third party extensions like
+
+    *   Fluid-Styled Content: :confval:`styles.templates.templateRootPath <typo3/cms-fluid-styled-content:fluid-styled-content-styles-templates-templaterootpath>`
+    *   Frontend Login: :ref:`felogin.email.templateRootPath <typo3/cms-felogin:confval-felogin-felogin-email-templaterootpath>`
+
+*   In some special cases paths might be set in PHP as well, for example when a
+    ViewFactory is used: `Using the generic view factory (ViewFactoryInterface) <https://docs.typo3.org/permalink/t3coreapi:generic-view-factory>`_.
+
+.. _extension-resources-private-fluid-plugin:
+
+Common locations for Fluid templates TYPO3 extensions with plugins:
+===================================================================
+
+..  typo3:file:: SomeTemplate.html
+    :scope: extension
+    :path: /Resources/Private/Templates/
+    :regex: /^.*\/Resources\/Private\/Templates\/[A-Za-z0-9]+\.html$/
+    :shortDescription: Fluid Template for an extension
+
+    Folder `Templates` often contains the Fluid templates for a TYPO3 extension.
```

Comment:
What about Extbase context where templates are stored in `Resources/Private/Templates/[ControllerName]/[ActionName].[format]`

Reply:
True, we should mention that

Reply:
is the .format thingy still working?

Reply:
Yes: https://github.com/TYPO3/typo3/blob/main/typo3/sysext/extbase/Classes/Mvc/Controller/ActionController.php#L510-L516

But [format] is not a must have for me. If it makes documentation cleaner I can sleep pretty good with just `.html`

Reply:
I added a tip and adjusted the regex